modules/apps/image/image-impl/src/main/java/com/liferay/image/internal/ImageMagickImpl.java

```diff
@@ -198,6 +201,55 @@ public void reset() {
 		}
 	}
 
+	@Override
+	public byte[] scale(byte[] bytes, String mimeType, int width, int height)
+		throws Exception {
+
+		if ((width == 0) && (height == 0)) {
+			return bytes;
+		}
+
+		File imageFile = null;
+		File scaledImageFile = null;
+
+		try {
+			imageFile = _file.createTempFile(bytes);
+
+			scaledImageFile = _file.createTempFile(mimeType);
+
+			List<String> arguments = new ArrayList<>();
+
+			arguments.add(imageFile.getAbsolutePath());
+			arguments.add("-resize");
+
+			if (height == 0) {
+				height = width;
+			}
+
+			if (width == 0) {
+				width = height;
+			}
```

Comment:
Previously, if `height` or `width` was zero, the whole `-resize` argument would have been skipped, but now it is always included (with a 0 value). Is this on purpose?

Reply:
Yes, it's on porpouse.

Not adding -resize produces this issue: https://liferay.atlassian.net/browse/LPS-187611

Remember that preview AM configuration is 1000x0.

In ImageMagick it doesn't seem to be problematic, because it respects the aspect ratio by default, so we could pass another value instead of 0 in this cases (I reuse the other parameter).

My first approach was trying too calculate this value based in the size of the image, as we do in com.liferay.portal.image.ImageToolImpl#scale(java.awt.image.RenderedImage, int):

```
@Override
	public RenderedImage scale(RenderedImage renderedImage, int width) {
		if (width <= 0) {
			return renderedImage;
		}

		int imageHeight = renderedImage.getHeight();

		int imageWidth = renderedImage.getWidth();

		double factor = (double)width / imageWidth;

		int scaledHeight = (int)Math.round(factor * imageHeight);

		int scaledWidth = width;

		return doScale(renderedImage, scaledHeight, scaledWidth);
	}
```
But generating the RenderedImage in some images types (for example webp) could be problematic so I simplify this.

Anyway, in https://liferay.atlassian.net/browse/LPS-187611 we have other proposal about how to solve this:

> I suggest fixing it like that
> 
> 
> ```
> if ((maxHeight > 0) && (maxWidth > 0)) {
> return StringBundler.concat(maxWidth, "x", maxHeight, ">");
> }
> if (maxWidth > 0) {
> return StringBundler.concat(maxWidth, ">");
> }
> if (maxHeight > 0) {
> return StringBundler.concat("x",maxWidth, ">");
> }
> ```

Reply:
Ok, sounds good.